src/Templates/src/templates/maui-mobile/.template.config/template.json

```diff
@@ -60,7 +60,7 @@
         "type": "parameter",
         "description": "Overrides the $(ApplicationId) in the project",
         "datatype": "string",
-        "replaces": "com.companyname.MauiApp.1"
+        "replaces": "com.companyname.mauiapp"
       },
```

Comment:
It's not clear to me from the docs on template.json, but could we add something like this:

```json
        "defaultValue": "com.companyname.MauiApp.1"
```

It seems _implied_ that you can do substitutions as part of the template metadata itself. If this works, I think it would get us the best of both worlds: if you specify a custom appName, you get exactly what you asked for. And if you don't say anything, you get a "fairly unique" name that is at least based on the project name you picked (or implied).